include/core/mediator/UnitsHelper.hpp

```diff
@@ -9,10 +9,12 @@ class UnitsHelper {
 
     public:
 
-    static double get_converted_value(const std::string &in_units, double value, const std::string &out_units);
+    static double get_converted_value(const std::string &in_units, const double &value, const std::string &out_units);
 
-    private:
+    static double* get_converted_values(const std::string &in_units, double* values, const std::string &out_units, const size_t & count);
```

Comment:
Do we want this to work on raw arrays. As defined this will change random memory if 'count' is not correctly set to the size of the array. Also returning a value is misleading as we are mutating the call argument.

Reply:
This is just a thin wrapper around the UDUNITS function that works on arrays `cv_convert_doubles(conv, values, count, values);`
```c++
double* UnitsHelper::get_converted_values(const std::string &in_units, double* values, const std::string &out_units, const size_t& count)
{
    if(in_units == out_units){
        return values; // Early-out optimization
    }
    std::call_once(unit_system_inited, init_unit_system);
    ut_unit* to = NULL;
    ut_unit* from = NULL;
    cv_converter* conv = get_converter(in_units, out_units, to, from);

    cv_convert_doubles(conv, values, count, values);
    ut_free(from);
    ut_free(to);
    cv_free(conv);
    return values;
}
```
And returning the value (reference to the mutated input variable) is just syntactic convenience and follows the potentially expected pattern as the regular `get_converted_value` which returns the converted value.  It could be said that NOT returning it is misleading in that context.  I implemented it this way to  be flexible, and it is a somewhat common pattern when working with mutable args behind a pointer.